src/rocker/bashrc_extension.py

```diff
@@ -0,0 +1,79 @@
+import em
+import pkgutil
+import os
+import urllib3
+from rocker.core import get_user_name, ExtensionError
+from rocker.extensions import RockerExtension
+
+
+class BashrcExtensions(RockerExtension):
+
+    name = 'bashrc_extensions'
+
+    @classmethod
+    def get_name(cls):
+        return cls.name
+
+    def __init__(self):
+        self._env_subs = None
+        self.name = BashrcExtensions.get_name()
+
+    def precondition_environment(self, cli_args):
+        pass
+
+    def validate_environment(self, cli_args):
+        pass
+
+    def get_preamble(self, cli_args):
+        return ''
+    
+    def get_filename(self, bashrc_extension_file):
+        if os.path.isfile(bashrc_extension_file):
+            return os.path.join(self.name, os.path.basename(bashrc_extension_file))
+        elif bashrc_extension_file.startswith(('http://', 'https://')):
+            filename = os.path.basename(urllib3.util.url.parse_url(bashrc_extension_file).path)
+            if not filename:
+                raise ExtensionError('Bashrc extension file does not appear to have a filename: {}'.format(bashrc_extension_file))
+            return os.path.join(self.name, filename)
+        else:
+            raise ExtensionError('Bashrc extension files is not a file or URL: {}'.format(bashrc_extension_file))
+
+    def get_files(self, cli_args):
+        files = {}
+        for bashrc_extension in cli_args[self.name]:
+            if os.path.isfile(bashrc_extension):
+                with open(bashrc_extension, 'r') as f:
+                    files[self.get_filename(bashrc_extension)] = f.read()
+            elif bashrc_extension.startswith(('http://', 'https://')):
+                try:
+                    response = urllib3.PoolManager().request('GET', bashrc_extension)
```

Comment:
This is a big change in scope. Fetching remote content opens up a whole extra can of worms. You can easily wget it to file just before running with barely more characters. Lets not add this level of complexity. Especially as this is a path to full remote code execution in the environment when you enter the container as we're not validating the remote content.